sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,244 @@
+        Parameters
+        ----------
+        timerange: `sunpy.time.TimeRange`
+            time range for which data is to be downloaded.
+        Returns
+        -------
+        urls : list
+            list of URLs corresponding to the requested time range
+        """
+        table_physobs = {'INTENSITY': 'i', 'LOS_MAGNETIC_FIELD': 'b'}
+        table_instruments = {'bigbear': 'bb', 'cerrotololo': 'ct', 'learmonth': 'le',
+                             'maunaloa': 'ml', 'teide': 'td', 'udaipur': 'ud', 'tucson': 'z'}
+
+        # Is PhysObs entered?
+        physobs_in = ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs.keys())
+        # Is instrument entered?
+        instrument_in = ('instrument' in kwargs.keys()
+                         and kwargs['instrument'] in table_instruments.keys())
+        wavelength_in = ('wavelength' in kwargs.keys())
+
+        url_pattern_1 = ('ftp://gong2.nso.edu/QR/{id}qa/%Y%m/{ObsID}{id}qa%y%m%d/'
+                         '{ObsID}{id}qa%y%m%dt%H%M.fits.gz')
+        url_pattern_2 = 'http://gong2.nso.edu/HA/haf/%Y%m/%Y%m%d/%Y%m%d%H%M%S{ObsID}h.fits.fz'
+
+        pattern_table = {6562: url_pattern_1, 6563: url_pattern_1, 6768: url_pattern_2}
+
+        result = list()  # final list of urls
+        patterns = list()
+        if not physobs_in:
+            patterns = [url_pattern_1, url_pattern_2]
+        else:
+            if kwargs['physobs'].lower() in ['intensity', 'los_magnetic_field']:
+                patterns.append(url_pattern_1)
+            else:
+                # Differentiate on basis of wavelength
+
+                # The isclose function in numpy can only check two arrays if they
+                # are close to each other within some precision. Standalone values such
+                # as int, doubles etc can't be checked that way. In order to do that, we put the
+                # two indiviual values in two seperate arrays da and db and then apply
+                # isclose on both those arrays.
+                if not wavelength_in:
+                    patterns.extend([url_pattern_1, url_pattern_2])
+                else:
+                    try:
+                        wave = kwargs['wavelength']
+                        wave_float = (
+                            wave.min if isinstance(
+                                wave, a.Wavelength) else wave.wavemin).value
+                        da = list()
+                        da.append(wave_float)
+                        for wave_nums in pattern_table.keys():
+                            db = list()
+                            db.append(wave_nums)
+                            if np.isclose(da, db, 1e-10, 1e-10):
+                                wave = wave_nums
+                                break
+                        patterns.append(pattern_table[wave])
+                    except BaseException:
+                        raise NameError("Enter correct wavelength range and units")
```

Comment:
Please, modify this `try` and `np.isclose` as suggested in #3810

Reply:
@dpshelio I have made the suggested changes.